java-components/cli/src/main/java/io/github/redhatappstudio/jvmbuild/cli/builds/BuildDiagnosticCommand.java

```diff
@@ -0,0 +1,122 @@
+        + "Dockerfiles for a build")
+public class BuildDiagnosticCommand
+        implements Runnable {
+
+    @CommandLine.Option(names = "-g", description = "The build to retrieve, specified by GAV", completionCandidates = GavCompleter.class)
+    String gav;
+
+    @CommandLine.Option(names = "-a", description = "The build to retrieve, specified by ArtifactBuild name", completionCandidates = ArtifactBuildCompleter.class)
+    String artifact;
+
+    @CommandLine.Option(names = "-b", description = "The build to retrieve, specified by build id", completionCandidates = BuildCompleter.class)
+    String build;
+
+    @CommandLine.Option(names = "-d", description = "Directory to save the Dockerfiles to. Defaults to current directory.", completionCandidates = BuildCompleter.class)
+    File targetDirectory = new File(System.getProperty("user.dir"));
+
+    @Override
+    public void run() {
+        var client = Arc.container().instance(KubernetesClient.class).get();
+        DependencyBuild theBuild = null;
+        if (build != null) {
+            if (artifact != null || gav != null) {
+                throwUnspecified();
+            }
+            Map<String, DependencyBuild> names = BuildCompleter.createNames();
+            theBuild = names.get(build);
+            if (theBuild == null) {
+                for (var n : names.values()) {
+                    if (build.equals(n.getMetadata().getName())) {
+                        //can also specify by kube name
+                        theBuild = n;
+                        break;
+                    }
+                }
+            }
+        } else if (artifact != null) {
+            if (gav != null) {
+                throwUnspecified();
+            }
+            ArtifactBuild ab = ArtifactBuildCompleter.createNames().get(artifact);
+            theBuild = BuildConverter.buildToArtifact(client, ab);
+        } else if (gav != null) {
+            ArtifactBuild ab = GavCompleter.createNames().get(gav);
+            theBuild = BuildConverter.buildToArtifact(client, ab);
+        } else {
+            throw new RuntimeException("Must specify one of -b, -a or -g");
+        }
+        if (theBuild == null) {
+            throw new RuntimeException("Build not found");
+        }
+        int lastSlash = theBuild.getSpec().getScm().getScmURL().lastIndexOf('/') + 1;
+        String name = theBuild.getSpec().getScm().getScmURL().substring(lastSlash).replaceFirst("\\.git.*", "");
+        System.out.println("Target directory: " + targetDirectory);
+        System.out.println(
+                "Selected build: " + theBuild.getMetadata().getName() + " of " + theBuild.getSpec().getScm().getScmURL() + ':' +
+                        theBuild.getSpec().getVersion() + '\n');
+
+        List<String> dockerFiles = theBuild.getStatus().getDiagnosticDockerFiles();
+        if ((theBuild.getStatus().getFailedBuildRecipes() == null ? 0 : theBuild.getStatus().getFailedBuildRecipes().size())
```

Comment:
This code does not work if the build has failed (which is generally the case when you are using this).